### ISSUE TITLE
setup: Apache, mod_wsgi , django with venv - error - bitsandbytes tries do find libcudart.so inside /var/log/apache folder.

### ISSUE THREAD

**Issue by decunde** [bitsandbytes-foundation/bitsandbytes#943]
### System Info

I am running in ubuntu, I set up the apache conf file to use python3.10 in a venv environment where I installed all packages for django and also all packages to run the model using GPU.

The error when trying to load the model / tokenizer is (found in apache error.log):
bitsandbytes denied permission to /var/log/apache/libcudart.so
the same program runs alone in ubuntu correctly with the same requirements, but when going by apache/mod_wsgi it tries to find the libs on the /var/log/path. I tried to set LD_LIBRAY_PATH and also change my wsg script, but bitsandbytes keeps looking for this apache log folder.

Below , my settings and requirements.

requirements.txt:
accelerate @ git+https://github.com/huggingface/accelerate.git@848ed800fac1a84e8fbda853086f06f5699835e5
aiohttp==3.9.1
aiosignal==1.3.1
asgiref==3.7.2
async-timeout==4.0.3
attrs==23.1.0
bitsandbytes==0.41.3.post2
certifi==2023.11.17
charset-normalizer==3.3.2
datasets==2.16.0
dill==0.3.7
Django==5.0
einops==0.7.0
filelock==3.13.1
frozenlist==1.4.1
fsspec==2023.10.0
huggingface-hub==0.20.1
idna==3.6
Jinja2==3.1.2
MarkupSafe==2.1.3
mpmath==1.3.0
multidict==6.0.4
multiprocess==0.70.15
networkx==3.2.1
numpy==1.26.2
nvidia-cublas-cu12==12.1.3.1
nvidia-cuda-cupti-cu12==12.1.105
nvidia-cuda-nvrtc-cu12==12.1.105
nvidia-cuda-runtime-cu12==12.1.105
nvidia-cudnn-cu12==8.9.2.26
nvidia-cufft-cu12==11.0.2.54
nvidia-curand-cu12==10.3.2.106
nvidia-cusolver-cu12==11.4.5.107
nvidia-cusparse-cu12==12.1.0.106
nvidia-nccl-cu12==2.18.1
nvidia-nvjitlink-cu12==12.3.101
nvidia-nvtx-cu12==12.1.105
packaging==23.2
pandas==2.1.4
peft @ git+https://github.com/huggingface/peft.git@cf04d0353f0343cbf66627228c4495f51669af34
psutil==5.9.7
pyarrow==14.0.2
pyarrow-hotfix==0.6
python-dateutil==2.8.2
pytz==2023.3.post1
PyYAML==6.0.1
regex==2023.12.25
requests==2.31.0
safetensors==0.4.1
scipy==1.11.4
sentencepiece==0.1.99
six==1.16.0
sqlparse==0.4.4
sympy==1.12
tokenizers==0.15.0
torch==2.1.2
tqdm==4.66.1
transformers @ git+https://github.com/huggingface/transformers.git@3cefac1d974db5e2825a0cb2b842883a628be7a0
triton==2.1.0
typing_extensions==4.9.0
tzdata==2023.3
urllib3==2.1.0
xxhash==3.4.1
yarl==1.9.4

apache conf:
<VirtualHost *:80>
    #ServerName example.com  # Replace with your domain or IP
    #ServerAdmin webmaster@localhost
    DocumentRoot /var/www/aiexp/aiprj/aiprj/static

    WSGIDaemonProcess aiprj python-home=/var/www/aiexp/venv python-path=/var/www/aiexp/aiprj
    WSGIScriptAlias / /var/www/aiexp/aiprj/aiprj.wsgi
    WSGIProcessGroup airprj
    WSGIApplicationGroup %{GLOBAL}

    <Directory /var/www/aiexp/aiprj/aiprj>
        <Files wsgi.py>
            Require all granted
        </Files>
    </Directory>

    Alias /static /var/www/aiexp/aiprj/staticfiles
    <Directory /var/www/aiexp/aiprj/staticfiles>
        Require all granted
    </Directory>

    ErrorLog ${APACHE_LOG_DIR}/error.log
    CustomLog ${APACHE_LOG_DIR}/access.log combined
</VirtualHost>

### Reproduction

    def loadModel(self):

        if self.device == 'cpu':
            self.model = AutoModelForCausalLM.from_pretrained(
                "Open-Orca/Mistral-7B-OpenOrca")
        else:
                nf4_config = BitsAndBytesConfig(
                load_in_4bit=True,
                bnb_4bit_quant_type="nf4",
                bnb_4bit_use_double_quant=True,
                bnb_4bit_compute_dtype=torch.float16
                )
                self.model = AutoModelForCausalLM.from_pretrained(
                "Open-Orca/Mistral-7B-OpenOrca",device_map='auto',quantization_config=nf4_config)

        self.tokenizer = AutoTokenizer.from_pretrained("Open-Orca/Mistral-7B-OpenOrca",padding_side="left")

        # Freeze the entire model
        for param in self.model.parameters():
            param.requires_grad = False

        # max_length=256  #early_stopping=False
        self.generation_config = GenerationConfig(
            max_length=num_t, max_new_tokens=num_t, temperature=num_tem, top_p=num_top, repetition_penalty=1num+pen,
            do_sample=True, use_cache=True,early_stopping=False,
            eos_token_id=self.tokenizer.eos_token_id, pad_token_id=self.tokenizer.eos_token_id,
            transformers_version="4.34.0.dev0")

        return 'loaded'

### Expected behavior

shoud load the model in memory

**Comment by Titus-von-Koeller** [comment]
closing as stale

**Comment by Titus-von-Koeller** [comment]
comment for refactor: BNB shouldn't search for CUDA in `/var/log/`

**Comment by TimDettmers** [comment]
Closing this issue. The old CUDA detection system in bitsandbytes 0.41.3 searched all PATH-like environment variables for CUDA libraries, which led to it scanning inappropriate locations like `/var/log/apache`. That entire module was replaced in v0.43.0 with a mechanism that uses PyTorch's CUDA detection instead. Please upgrade to the latest version.
